components/brave_rewards/core/database/database_sku_transaction.h

```diff
@@ -9,11 +9,13 @@
 #include <string>
 
 #include "brave/components/brave_rewards/core/database/database_table.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
 
 namespace ledger {
 namespace database {
 
-using GetSKUTransactionCallback = std::function<void(mojom::SKUTransactionPtr)>;
+using GetSKUTransactionCallback =
+    std::function<void(absl::optional<mojom::SKUTransactionPtr>)>;
```

Comment:
The distinction between the failure case and the "no records returned" case seems confusing with this signature (essentially, `optional of optional of x`).

We ran into the same thing with the external transaction queries. Perhaps we should use the same pattern used over there? Other there it's:

```
enum class GetExternalTransactionError { kDatabaseError, kTransactionNotFound };

base::expected<mojom::ExternalTransactionPtr, GetExternalTransactionError>
```